packages/utils/src/bigint.ts

```diff
@@ -94,3 +82,26 @@ export const formatUnits = (value: bigint, decimals: number = 18): string => {
 }
 
 export const formatEther = (value: bigint): string => formatUnits(value, 18)
+
+// JSON.stringify doesn't handle BigInts, so we need to replace them with objects
+export const bigintReplacer = (key: string, value: any): any => {
+  if (typeof value === 'bigint') {
+    return { $bigint: value.toString() }
+  }
+
+  return value
+}
+
+// JSON.parse will need to convert our serialized bigints back into BigInt
+export const bigintReviver = (key: string, value: any): any => {
+  if (value !== null && typeof value === 'object' && '$bigint' in value && typeof value.$bigint === 'string') {
```

Comment:
Don't need `'$bigint' in value` since you're checking `typeof`:

```
if (value && typeof value === 'object' && typeof value.$bigint === 'string') {
```